vcd/datasource_vcd_nsxt_edgegateway.go

```diff
@@ -0,0 +1,127 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func datasourceVcdNsxtEdgeGateway() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdNsxtEdgeGatewayRead,
+		Schema: map[string]*schema.Schema{
+			"org": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Edge Gateway name",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Edge Gateway description",
+			},
+			"dedicate_external_network": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Dedicating the External Network will enable Route Advertisement for this Edge Gateway.",
+			},
+			"external_network_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "External network ID",
+			},
+			"subnet": {
+				Description: "One or more blocks with external network information to be attached to this gateway's interface",
+				Computed:    true,
+				Type:        schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"gateway": {
+							Computed:    true,
```

Comment:
would be nice to follow format of 
```
	                         Type:        schema.TypeString,
				Computed:    ...,
				Description: "....",
````
In subtypes

Reply:
Agreed.. Fixed for datasources and resource.